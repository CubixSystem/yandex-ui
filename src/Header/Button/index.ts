export * from './_view/Button_view_search-arrow';
