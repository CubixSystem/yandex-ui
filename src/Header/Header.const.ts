import { cn } from '@bem-react/classname';

export const cnHeader = cn('YandexHeader');
export const cnHeaderBase = cnHeader('Base');
export const cnHeaderLogoWrapper = cnHeader('LogoWrapper');
export const cnHeaderLogo = cnHeader('Logo');
export const cnHeaderLogoInline = cnHeader('Logo-Inline');
export const cnHeaderContent = cnHeader('Content');
export const cnHeaderIcon = cnHeader('Icon');
export const cnHeaderInlineIcon = cnHeader('InlineIcon');
export const cnHeaderImage = cnHeader('Image');
export const cnHeaderActions = cnHeader('Actions', [cnHeader('Item')]);
