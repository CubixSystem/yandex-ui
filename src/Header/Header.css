.YandexHeader {
    position: relative;
    z-index: var(--header-z-index);

    padding: var(--header-space-all);

    font-family: var(--header-font-family);

    background: var(--header-fill-color);
    box-shadow: var(--header-shadow);
}

.YandexHeader-Base {
    display: flex;
    justify-content: space-between;
    align-items: center;

    box-sizing: border-box;
    max-width: var(--header-content-max-width);
    height: var(--header-content-height);
    margin: auto;
    padding: var(--header-content-space-all);
}

.YandexHeader-Content {
    display: inline-flex;
    align-items: center;

    width: var(--header-content-width);
    margin-right: var(--header-content-indent-right);
}

.YandexHeader-LogoWrapper {
    display: inline-flex;
    align-items: inherit;

    margin-right: var(--header-logo-indent-right);

    white-space: nowrap;
}

.YandexHeader-Actions {
    display: inline-flex;
    flex-shrink: 0;
    justify-content: flex-end;
    align-items: center;

    padding-left: var(--header-actions-space-left);
}

.YandexHeader-Actions > * + * { /* stylelint-disable-line selector-max-universal */
    margin-left: var(--header-actions-gap);
}
