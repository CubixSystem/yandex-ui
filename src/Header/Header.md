# Header

Подробнее про компонент в <a href="https://lego.yandex-team.ru/lego-components/components/header/usage" target="_blank">документации</a>.
