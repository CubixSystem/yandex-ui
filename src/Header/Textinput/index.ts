export * from './_view/Textinput_view_search-arrow';
