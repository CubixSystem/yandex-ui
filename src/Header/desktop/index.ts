export * from '../Header';
export * from '../Header.const';
export * from '../Nav/Header-Nav';
export * from '../Search/Header-Search';
export * from '../Logo/Header-Logo';
export * from '../Logo/Inline-Logo';
export * from '../Logo/Logoaas';
export * from '../Icons/Header-Icons';
export * from '../Icons/Mirror-Icon';
export * from '../Icons/Weather-Icon';
export * from '../Icons/Sport-Icon';
export * from '../Icons/News-Icon';
export * from '../Icons/Services-Icon';
export * from '../Icons/Inline-Icons';
