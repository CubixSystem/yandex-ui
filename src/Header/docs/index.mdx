---
id: Header
path: /components/Header

title: Header
description: Компонент шапки.

links:
 - label: source
   url: https://a.yandex-team.ru/arc_vcs/frontend/packages/lego-components/src/Header
 - label: startrek
   url: https://search.yandex-team.ru/stsearch?text=Header.ts&facet.queue=ISL&facet.type=bug&facet.status=128
 - label: storybook
   url: https://lego.yandex-team.ru/components/lego/latest/?storybook=%2Fstory%2Flayout-header-desktop--playground

tabs:
 - Usage
 - Structure
 - Examples
 - Migration
---
