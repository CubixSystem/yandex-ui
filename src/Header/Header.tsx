import React, { FC, ReactNode, Ref } from 'react';

import { cnHeader, cnHeaderBase, cnHeaderLogoWrapper, cnHeaderContent, cnHeaderActions } from './Header.const';
import { YandexLogo } from './Logo/Header-Logo';
import './Header.css';

export type HeaderProps = {
    /**
     * Дополнительный класс на DOM элемент.
     */
    className?: string;

    /**
     * Ссылка на корневой DOM-элемент компонента
     */
    innerRef?: Ref<HTMLDivElement>;

    /**
     * Полностью свой логотип.
     */
    logo?: ReactNode;

    /**
     * Action элементы в шапке.
     */
    actions?: ReactNode;
};

/**
 * Компонент шапки.
 * @param {HeaderProps} props
 *
 */
export const Header: FC<HeaderProps> = ({ actions, children, className, innerRef, logo }) => {
    return (
        <header ref={innerRef} className={cnHeader(null, [className])}>
            <div className={cnHeaderBase}>
                <div className={cnHeaderLogoWrapper}>{logo || <YandexLogo />}</div>
                <div className={cnHeaderContent}>{children}</div>
                {actions && <div className={cnHeaderActions}>{actions}</div>}
            </div>
        </header>
    );
};

Header.displayName = 'Header';
