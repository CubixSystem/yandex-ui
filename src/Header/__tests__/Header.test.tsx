import React, { createRef } from 'react';

import { createClientRender, screen, cleanup } from '../../internal/testing';
import { Header } from '../Header';

describe('Header', () => {
    const render = createClientRender();

    test('should render correctly', () => {
        const { container } = render(<Header />);
        expect(container).toMatchSnapshot();
    });

    test('should return correct displayName', () => {
        expect(Header.displayName).toBe('Header');
    });

    test('should set innerRef', () => {
        const innerRef = createRef<HTMLDivElement>();
        render(<Header innerRef={innerRef} />);
        expect(innerRef.current).not.toBe(null);
    });

    test('should set actions', () => {
        render(<Header />);
        const text = 'actions';
        try {
            screen.getByText(text);
        } catch (e) {
            expect(e.message).toContain(`Unable to find an element with the text: ${text}`);
        }
        cleanup();
        render(<Header actions={text} />);
        expect(screen.getByText(text)).not.toBe(null);
    });

    test('should set children', () => {
        const text = 'children';
        render(<Header>{text}</Header>);
        try {
            screen.getByText(text);
        } catch (e) {
            expect(e.message).toContain(`Unable to find an element with the text: ${text}`);
        }
        cleanup();
        render(<Header>{text}</Header>);
        expect(screen.getByText(text)).not.toBe(null);
    });

    test('should set logo', () => {
        const text = 'logo';
        render(<Header />);
        try {
            screen.getByText(text);
        } catch (e) {
            expect(e.message).toContain(`Unable to find an element with the text: ${text}`);
        }
        cleanup();
        render(<Header logo={text} />);
        expect(screen.getByText(text)).not.toBe(null);
    });
});
