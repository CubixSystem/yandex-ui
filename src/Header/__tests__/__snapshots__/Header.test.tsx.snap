// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <header
    class="YandexHeader"
  >
    <div
      class="YandexHeader-Base"
    >
      <div
        class="YandexHeader-LogoWrapper"
      >
        <a
          class="Link YandexHeader-Logo YandexHeader-Logo_lang_ru"
          href="//yandex.ru"
          tabindex="-1"
        />
      </div>
      <div
        class="YandexHeader-Content"
      />
    </div>
  </header>
</div>
`;
